### PR TITLE
feat(GQL): RHICOMPL-2995 filter by unsupported/unreported hosts

### PR DESCRIPTION
**Pull request by skateman**
The `getSystems` query with `testResultProfiles` can now handle filtering by `supported_ssg` or `reported` boolean values.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by skateman** [comment]
```
{ 
  "operationName": "getSystems",
  "variables": {"policyId": "1646371b-1a24-4d65-935f-99986f007236", "filter": "policy_id = 1646371b-1a24-4d65-935f-99986f007236 and reported = false"},
  "query": "query getSystems($filter: String!, $policyId: ID, $perPage: Int, $page: Int, $sortBy: [String!], $tags: [String!]) {
  systems(
    search: $filter
    limit: $perPage
    offset: $page
    sortBy: $sortBy
    tags: $tags
  ) {
    totalCount
    edges {
      node {
        id
        name
        osMajorVersion
        osMinorVersion
        culledTimestamp
        staleWarningTimestamp
        staleTimestamp
        insightsId
        lastScanned
        updated
        testResultProfiles(policyId: $policyId) {
          id
          name
          refId
          lastScanned
          compliant
          external
          score
          supported
          ssgVersion
          osMajorVersion
          rules {
            refId
            title
            compliant
            remediationAvailable
            precedence
            __typename
          }
          __typename
        }
        policies(policyId: $policyId) {
          id
          name
          __typename
        }
        tags {
          namespace
          key
          value
          __typename
        }
        __typename
      }
      __typename
    }
    __typename
  }
}
```

**Comment by InsightsDroid** [comment]
Test results are available in Ibutsu:
https://url.corp.redhat.com/ibutsu-runs-c117a25d-6f78-40bc-87d2-2440e7fdc85f

**Review by romanblanco — APPROVED** [review]
LGTM :+1:

File: app/models/concerns/host_searching.rb [hunk 1]
```diff
@@ -32,6 +32,12 @@ module HostSearching
                   only_explicit: true, operators: ['=']
     scoped_search on: :stale_timestamp,
                   only_explicit: true, operators: ['<', '>']
+    scoped_search on: :supported,
```
Comment (vkrizan) [inline app/models/concerns/host_searching.rb:35]: To be more specific, can it be named `ssg_supported`? `supported` sounds very generic.

**Pull request merged**